src/Features/Core/Portable/ConvertForEachToFor/AbstractConvertForEachToForCodeRefactoringProvider.cs

```diff
@@ -169,11 +169,6 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 }
             }
 
-            if (CheckIfForEachVariableIsWrittenInside(model, foreachVariable, foreachStatement))
```

Comment:
>CheckIfForEachVariableIsWrittenInside [](start = 16, length = 37)

In VB it is legal to assign to the iteration variable, so this check is probably useful.

Reply:
My thinking is that if we don't check if the variable is written for C# and it's legal in VB to write to it then the data flow analysis is basically a waste of cpu cycles as the results will never have any effect on the outcome.
Unless I'm missing something?

Reply:
@Paxxi the check is there so that we don't accidently change semantic of code. sure, we can relax the check if we feel it is too restrictive. or we could bailout on the check if the variable is struct.

Reply:
another option is to have a warning in the code-action in this case.  that's something we already support.

Reply:
When rewriting the `foreach` it's rewritten in this form
```C#
for (var i = 0; i < test.Length; i++)
{
    var foo = test[i];
    _ = foo.SomeProperty;
}
``` from https://github.com/dotnet/roslyn/issues/32739#issuecomment-466683307 .

My understanding is that with this method there should be no change in semantics?

If I'm keeping flow analysis and add the warning, any suggestions on wording?

Reply:
"Warning: Semantics may change when converting this 'foreach' statement."

Reply:
yep. warning works as well.

Reply:
I could use a hint or two here.
I don't believe I've seen this warning support used anywhere so not sure where to start looking.
Did try to poke around in the code but didn't find anything that seemed appreciate.

Reply:
@Paxxi - did you still need guidance on how to add this warning annotation to the code action preview? You can look at http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/CodeActions/Annotations/WarningAnnotation.cs,aabcf60b526f2d32,references for some existing usages.

Reply:
Thank you! I'll take a look at it again this week

Reply:
New attempt at fixing this is pushed now 😄